ptgnn/implementations/typilus/train.py

```diff
@@ -160,8 +162,20 @@ def run(arguments):
         nn = None
         model = create_graph2class_gnn_model()
 
-    def create_optimizer(parameters):
-        return torch.optim.Adam(parameters, lr=0.00025)
+    if arguments["--restore-optimizer"]:
+        opt_state = torch.load(arguments["--restore-optimizer"])
+
+        def create_optimizer(parameters):
+            opt = torch.optim.Adam(parameters, lr=0.00025)
+            logging.info("Restoring optimizer state from `%s`.", arguments["--restore-optimizer"])
+
+            opt.load_state_dict(opt_state["optimizer_state_dict"])
+            return opt
+
```

Comment:
To resume training, it should load the index of start epoch.
```suggestion
start_epoch_idx = opt_state["epoch"]
```

And then need to change from `for epoch in range(self._max_num_epochs):` to `for epoch in range(start_epoch_idx, self._max_num_epochs):` in both distributedtrainer.py and trainer.py. You can set `start_epoch_idx = 0` as default if `arguments["--restore-optimizer"]` is None.

Reply:
now implemented, although not absolutely necessary to restart the epoch_idx from the previous one (matters only for schedulers)